geoprocessing/src/modules/shapefiles/shapefiles.service.ts

```diff
@@ -42,25 +19,20 @@ export class ShapeFileService {
     return JSON.parse(_geoJson[outputKey].toString('utf-8'));
   }
 
-  private deleteShapefileData(path: string): void {
-    if (path.startsWith('/tmp')) {
-      unlinkSync(path);
-      rmdirSync(path.replace('.zip', ''), { recursive: true });
-    } else {
-      throw new Error(`Could not complete deletion: ${path} is not in /tmp`);
-    }
-  }
-
   async getGeoJson(shapeFile: Express.Multer.File) {
     try {
-      this.logger.log(await this.unzipShapefile(shapeFile));
+      this.logger.log(await this.fileService.unzipFile(shapeFile));
     } catch (err) {
       this.logger.error(err);
     }
-    const geoJson = await this.shapeFileToGeoJson(shapeFile).then((geoJson) => {
-      this.deleteShapefileData(shapeFile.path);
-      return geoJson;
-    });
+    const geoJson = await this.shapeFileToGeoJson(shapeFile).then(
+      async (geoJson) => {
+        await this.fileService
+          .deleteDataFromFS(shapeFile.path)
+          .catch((error) => error);
+        return geoJson;
+      },
+    );
```

Comment:
it's a bit hard to read and seems to be unnecessarily nested, I'd suggest this:
```suggestion
    const geoJson = await this.shapeFileToGeoJson(shapeFile);
    await this.fileService
      .deleteDataFromFS(shapeFile.path)
      .catch((error) => error);
```

Also, `catch((error) => error);` is not the best option, as we will miss issues with file storage. I would at least try to log it, preferably with an `error` log level with some context of the error.

Reply:
I agree 7c1a76eee6b4e7692467d485d9cae6b487d3d8b0